playground/physx-controller.ts

```diff
@@ -0,0 +1,485 @@
+        .then((res) => {
+          const materials = asset.materials;
+          for (let i = 0, n = materials.length; i < n; i++) {
+            const material = materials[i];
+            (<PBRMaterial>material).roughnessMetallicTexture = res;
+          }
+        });
+      engine.resourceManager
+        .load<Texture2D>("https://gw.alipayobjects.com/zos/OasisHub/440001585/2547/T_Doggy_1_ao.png")
+        .then((res) => {
+          const materials = asset.materials;
+          for (let i = 0, n = materials.length; i < n; i++) {
+            const material = materials[i];
+            (<PBRMaterial>material).occlusionTexture = res;
+          }
+        });
+      engine.resourceManager
+        .load<GLTFResource>("https://gw.alipayobjects.com/os/OasisHub/440001585/7205/Anim_Run.gltf")
+        .then((res) => {
+          const animations = res.animations;
+          if (animations) {
+            animations.forEach((clip: AnimationClip) => {
+              const animatorState = animatorStateMachine.addState(clip.name);
+              animatorState.clip = clip;
+            });
+          }
+        });
+      engine.resourceManager
+        .load<GLTFResource>("https://gw.alipayobjects.com/os/OasisHub/440001585/3380/Anim_Idle.gltf")
+        .then((res) => {
+          const animations = res.animations;
+          if (animations) {
+            animations.forEach((clip: AnimationClip) => {
+              const animatorState = animatorStateMachine.addState(clip.name);
+              animatorState.clip = clip;
+            });
+          }
+          animator.play("Idle");
+        });
+      engine.resourceManager
+        .load<GLTFResource>("https://gw.alipayobjects.com/os/OasisHub/440001585/5703/Anim_Landing.gltf")
+        .then((res) => {
+          const animations = res.animations;
+          if (animations) {
+            animations.forEach((clip: AnimationClip) => {
+              const animatorState = animatorStateMachine.addState(clip.name);
+              animatorState.clip = clip;
+            });
+          }
+        });
+      engine.resourceManager
+        .load<GLTFResource>("https://gw.alipayobjects.com/os/OasisHub/440001585/3275/Anim_Fall.gltf")
+        .then((res) => {
+          const animations = res.animations;
+          if (animations) {
+            animations.forEach((clip: AnimationClip) => {
+              const animatorState = animatorStateMachine.addState(clip.name);
+              animatorState.clip = clip;
+            });
+          }
+        });
+      engine.resourceManager
+        .load<GLTFResource>("https://gw.alipayobjects.com/os/OasisHub/440001585/2749/Anim_Jump_In.gltf")
+        .then((res) => {
+          const animations = res.animations;
+          if (animations) {
+            animations.forEach((clip: AnimationClip) => {
+              const animatorState = animatorStateMachine.addState(clip.name);
+              animatorState.clip = clip;
+            });
+          }
+        });
```

Comment:
Many assets load and organize code, which will interfere with the logical reading of developers

We can package the asset loading and organization logic as an independent function